src/pages/dev/general-message-passing/overview.mdx

```diff
@@ -21,14 +21,52 @@ With GMP you can:
 #### At the source chain
 
 1. User (dApp) calls a `callContract` (or `callContractWithToken`) function on the Axelar Gateway contract to initiate a call. Once the call is initiated, the user can see its status at https://axelarscan.io/gmp/[txHash] or programmatically track it via the [AxelarJS SDK](/dev/axelarjs-sdk/tx-status-query-recovery#query-transaction-status-by-txhash).
-2. User prepays the relayer gas fee on the source chain to Axelar’s Gas Services contract.
+2. The user prepays the relayer gas fee on the source chain to Axelar's Gas Services contract.
 3. The call enters the Axelar Gateway from the source chain.
 
 #### At the Axelar network
-4. Axelar network confirms the call and converts the paid gas from the source chain’s native token into the destination chain’s native token.  
+4. The Axelar network confirms the call and converts the paid gas from the source chain's native token into the destination chain's native token.  
 
 #### At the destination chain
 5. The call is approved and emerges from the Axelar Gateway on the destination chain.
-6. The executor service relays and executes the approved call to the application’s Axelar Executable interface.
+6. The executor service relays and executes the approved call to the application's Axelar Executable interface.
 
-Suppose the paid gas (step 2) is insufficient to relay the transfer to the application interface (step 6). We have [monitoring and recovery](/dev/general-message-passing/monitoring) steps to help deal with such scenarios.
\ No newline at end of file
+Suppose the paid gas (step 2) is insufficient to relay the transfer to the application interface (step 6). We have [monitoring and recovery](/dev/general-message-passing/monitoring) steps to help deal with such scenarios.
+
+
+### Real-World Breakdown of Cross-Chain [Contract Calls](https://axelarscan.io/gmp/0x93cb0b614b07d6050b164cc3e35da617a2fbefc13069a35369894cac74b861a2:54) 
+
+![simple-gmp-diagram.png](/images/simple-gmp-diagram.png)
+
+#### Setup
+
+a. The destination application contract implements the `IAxelarExecutable.sol` interface to receive the cross-chain message.
+
+b. The destination application contract stores the address of the destination chain's Axelar gateway contract.
```

Comment:
If we're mentioning this point for the dest contract should we not also mention it for the src contract? (which also needs the address of the gateway contract)

Reply:
This is the setup step as referenced [here](https://github.com/axelarnetwork/cgp-spec/tree/main/solidity#cross-chain-contract-call-flow) and happens on the destination contract

Reply:
Maybe simpler language "stores the address of the local Gateway contract"